test/testvm.py

```diff
@@ -971,15 +976,34 @@ def wait_reboot(self):
             raise Failure("system didn't reboot properly")
         self.wait_user_login()
 
-    def wait_boot(self, wait_for_running_timeout = 120):
+    def wait_boot(self, wait_for_running_timeout = 120, allow_one_reboot=False):
```

Comment:
I see this differently. In case of regular boot, https://github.com/cockpit-project/cockpit/pull/3261/files#diff-ad31112db9347f7ac1d8207975a36a5dR990 will connect and the code will wait for user login. If that succeeds, the function returns without going into the reboot specific code.